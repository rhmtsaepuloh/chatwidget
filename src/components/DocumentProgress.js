/* eslint-disable react/jsx-no-target-blank */
/* eslint-disable jsx-a11y/img-redundant-alt */
import React from "react";
import {
  FaFilePdf,
  FaFileExcel,
  FaRegTimesCircle,
  FaFileAlt,
} from "react-icons/fa";
import { MdInsertDriveFile } from "react-icons/md";
import RenderDocumentProgress from "./RenderDocumentProgress";

const DocumentProgress = ({
  extension,
  color,
  progressUpload,
  sourceImage,
  timestamp,
  text,
  caption,
  cancelUpload,
}) => {
  const SwitchDocument = () => {
    switch (extension) {
      case "pdf":
        return (
          <RenderDocumentProgress
            text={text}
            progressUpload={progressUpload}
            timestamp={timestamp}
            cancelUpload={cancelUpload}
            caption={caption}
            icon={<FaFilePdf style={{ color }} />}
          />
        );
      case "doc":
      case "docx":
        return (
          <RenderDocumentProgress
            text={text}
            progressUpload={progressUpload}
            timestamp={timestamp}
            cancelUpload={cancelUpload}
            caption={caption}
            icon={<MdInsertDriveFile style={{ color }} />}
          />
        );
      case "txt":
        return (
          <RenderDocumentProgress
            text={text}
            progressUpload={progressUpload}
            timestamp={timestamp}
            cancelUpload={cancelUpload}
            caption={caption}
            icon={<FaFileAlt style={{ color }} />}
          />
        );
      case "xls":
      case "xlsx":
        return (
          <RenderDocumentProgress
            text={text}
            progressUpload={progressUpload}
            timestamp={timestamp}
            cancelUpload={cancelUpload}
            caption={caption}
            icon={<FaFileExcel style={{ color }} />}
          />
        );
      default:
        return (
          <React.Fragment>
            <img
              src={sourceImage}
              alt='image'
              className='h-56 w-full mb-2 rounded-md'
            />
            <div className='file-image flex items-center'>
              <div className='progress bg-white' style={{ width: `${100}%` }}>
                <div
                  className='progress bg-blue-700'
                  style={{ width: `${progressUpload}%` }}></div>
              </div>
              <p className='text-blue-700 text-xs font-bold mx-1'>
                {progressUpload}%
              </p>
              <FaRegTimesCircle
                className='cursor-pointer'
                onClick={() => cancelUpload()}
              />
            </div>
            <p className='whitespace-pre-wrap break-words'>{caption}</p>
          </React.Fragment>
        );
    }
  };
  return SwitchDocument();
};

export default DocumentProgress;
