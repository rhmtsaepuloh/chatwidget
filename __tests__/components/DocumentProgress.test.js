/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import DocumentProgress from "../../src/components/DocumentProgress";

const props = {
  color: "#000",
  timestamp: "10:21",
  text: "kirim file",
  size: 781076,
  caption: "tambahkan caption",
  progressUpload: 80,
  sourceImage:
    "https://www.majalahict.com/wp-content/uploads/2020/05/Logo-ChatAja.jpg",
  cancelUpload: jest.fn(),
};

describe("render correctly document progress components", () => {
  test("renders if not exist extension", () => {
    const extension = null;
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file image", () => {
    const extension = "png";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file pdf", () => {
    const extension = "pdf";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file xls", () => {
    const extension = "xls";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file xlsx", () => {
    const extension = "xlsx";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file doc", () => {
    const extension = "doc";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file docx", () => {
    const extension = "docx";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders file txt", () => {
    const extension = "file.txt";
    const wrapper = shallow(
      <DocumentProgress {...props} extension={extension} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
