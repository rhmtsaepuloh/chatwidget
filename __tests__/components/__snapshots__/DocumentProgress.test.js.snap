// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly document progress components renders file doc 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file docx 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file image 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file pdf 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file txt 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file xls 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders file xlsx 1`] = `ShallowWrapper {}`;

exports[`render correctly document progress components renders if not exist extension 1`] = `ShallowWrapper {}`;
